pybop/parameters/parameter.py

```diff
@@ -149,9 +152,23 @@ def set_bounds(self, bounds=None):
             else:
                 self.lower_bound = bounds[0]
                 self.upper_bound = bounds[1]
+        elif self.prior is not None:
+            self.lower_bound = self.prior.mean - boundary_multiplier * self.prior.sigma
+            self.upper_bound = self.prior.mean + boundary_multiplier * self.prior.sigma
+            bounds = [self.lower_bound, self.upper_bound]
```

Comment:
Please add an output message here, to warn users that bounds have been applied - similar to the warning when bounds are not being applied.